src/reader.rs

```diff
@@ -606,3 +602,37 @@ pub fn item_leaf<'a, D: Distance>(
         None => Ok(None),
     }
 }
+
```

Comment:
```suggestion

// Based on https://quickwit.io/blog/top-k-complexity, implemented in https://github.com/meilisearch/arroy/pull/129
```

I would like to keep the source for the future me ahah